src/main/java/frc/robot/subsystems/elevator/ElevatorIOSim.java

```diff
@@ -1,15 +1,50 @@
 package frc.robot.subsystems.elevator;
 
+import static edu.wpi.first.units.Units.Meters;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.units.Distance;
 import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.wpilibj.Timer;
+import lib.motors.TalonFXSim;
 
 public class ElevatorIOSim implements ElevatorIO {
+    private final TalonFXSim motor;
+    private final ElevatorInputs inputs;
+
+    private final PositionVoltage positionRequest = new PositionVoltage(0);
+    private final DutyCycleOut powerRequest = new DutyCycleOut(0);
+
+    public ElevatorIOSim(ElevatorInputs inputs) {
+        this.inputs = inputs;
+        motor = new TalonFXSim(2, ElevatorConstants.GEAR_RATIO, 0.5, 1);
+
+        motor.setController(
+                new PIDController(
+                        ElevatorConstants.KP, ElevatorConstants.KI, ElevatorConstants.KD, 0.02));
+    }
+
     @Override
-    public void setPower(double power) {}
+    public void setPower(double power) {
+        powerRequest.withOutput(power);
+        motor.setControl(powerRequest);
+    }
 
     @Override
-    public void setHeight(MutableMeasure<Distance> height) {}
+    public void setHeight(MutableMeasure<Distance> height) {
+        positionRequest.withPosition(height.in(Meters));
+        motor.setControl(positionRequest);
+    }
 
     @Override
-    public void updateInputs(ElevatorInputs inputs) {}
+    public void updateInputs(ElevatorInputs inputs) {
+        motor.update(Timer.getFPGATimestamp());
+        inputs.currentHeight =
+                Meters.of(
+                                lib.units.Units.rpsToMetersPerSecond(
```

Comment:
Better to configure the ratio instead of calculating it yourself
https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/configs/FeedbackConfigs.html#withSensorToMechanismRatio(double)

Reply:
Agreed. Notice that the `TalonFXSim` constructor has a parameter called `conversionFactor`, which works exactly the same as the real motor conversion factor.

Reply:
I keep forgetting this is your class and not Phoenix's sim classes lol

Reply:
Changed it a long time ago